tools/azure-sdk-tools/ci_tools/build.py

```diff
@@ -189,3 +189,21 @@ def create_package(
 
     if enable_sdist:
         run([sys.executable, "setup.py", "sdist", "-d", dist], cwd=setup_parsed.folder, check=True)
+
+        # The file normalization changes in setuptools 67.3.0 intends source distribution file names to have the
+        # formatting a whl distributions. This patch avoids that for now.
+        normalized_package_name = setup_parsed.name.replace("-", "_")
```

Comment:
Isn't this reversed? I thought we are getting names that had `_` instead of `-`? Maybe I'm not understanding where all the names come from.

Do we think there will be any impact with renaming the actual files? We could also probably fix our package info script https://github.com/Azure/azure-sdk-for-python/blob/main/eng/scripts/get_package_properties.py#L23 to normalize to always have `-`.

Reply:
I don't think the issue is parsing the metadata from the setup.py. I'm just pinning the setuptools and then crawling the package_properties to try to understand where it's coming from.

Reply:
I'm going to pin setuptools to unblock folks for immediate action, but continue the pathway I'm currently on I think.

Reply:
Yeah. So when we're releasing a package, we pull the tag from the sdist distribution. We specifically used the sdist _because_ of the difference in formatting. It was easy to parse package name and version because we know the first `_` is the separator.

(`Language-Settings.ps1 line 93`)